service/host.go

```diff
@@ -199,7 +206,7 @@ func (s *host) AddModules(modules ...ModuleCreateFunc) error {
 		if !s.supportsRole(mi.Roles...) {
 			s.Logger().Info(
 				"module will not be added due to selected roles",
-				"roles", mi.Roles,
+				zap.Any("roles", mi.Roles),
```

Comment:
Not needed - `Any` handles all those cases. The only overhead is allocating to cast the slice to `interface{}`. They're all serialized properly, no %v needed. 

If you're okay with the one extra interface allocation, just using zap.Any for every field is a big perf win over the SugaredLogger. It's exactly what sugaring does under the hood - it's zipping the keys and values into Any.